frame/babe/src/lib.rs

```diff
@@ -216,7 +214,7 @@ impl<T: Trait> FindAuthor<u32> for Module<T> {
 	{
 		for (id, mut data) in digests.into_iter() {
 			if id == BABE_ENGINE_ID {
-				let pre_digest = RawPreDigest::decode(&mut data).ok()?;
+				let pre_digest: RawPreDigest = RawPreDigest::decode(&mut data).ok()?;
```

Comment:
Asking for my own learning: Why is the explicit type annotation necessary?

Reply:
It's due to the default type parameter of `RawPreDigest`. In expression they won't be auto-guessed, so if without the annotation it has to be explicit. We can also write:

```
let pre_digest = RawPreDigest::<RawVRFProof, RawVRFOutput>::decode(&mut data).ok()?;
```